Allow 7.1 audio for AAC

Nomis101 opened the pull request:
Currently AAC is limited to 5.1 channel layout. Allow 6.1 and 7.1 as well. I have tested this on macOS 10.13 and tvOS 11.

bradleysepos commented:
Please add .DS_Store to your (global) .gitignore.

Nomis101 commented:
Sorry, still learning. Why does it have to be so difficult on GitHub to submit patches. *sigh* How can I insert my edited request in here without opening a new pull request?
https://github.com/Nomis101/HandBrake/tree/AAC_7.1v2

bradleysepos commented:
You can overwrite your remote branch with `git push --force ...`.

jstebbins commented on libhb/common.c:
```diff
@@ -1790,12 +1790,14 @@ int hb_mixdown_has_codec_support(int mixdown, uint32_t codec)
         case HB_ACODEC_FFFLAC24:
         case HB_ACODEC_OPUS:
+        case HB_ACODEC_CA_AAC:
+        case HB_ACODEC_FDK_AAC:
```
FDK AAC doesn't support 6.1.  So it needs to be separated out as it's own case and return FALSE for 6.1 and TRUE for everything else. Also FDK HE-AAC (HB_ACODEC_FDK_HAAC) supports the same as FDK AAC.

Nomis101 commented:
I've updated the patch to show only 6.1 if AAC (CoreAudio) is selected and don't show 6.1 if FDK AAC is selected. Hope this is what you had in mind.

bradleysepos commented:
You will want to do:

```
git rebase -i 9259d7d~1
# mark all old and revert commits as drop
git push --force <remote> <branch>
```

That should clean up the commit history nicely.

jstebbins commented on libhb/common.c:
```diff
@@ -1939,13 +1942,23 @@ int hb_mixdown_get_default(uint32_t codec, uint64_t layout)
+        // FDK AAC doesn't support 6.1, separate it out as it's own case
+        case HB_ACODEC_FDK_AAC:
+        case HB_ACODEC_FDK_HAAC:
+            mixdown = HB_AMIXDOWN_5POINT1 ||
```
Nope.  This function just returns the default mixdown which should be the best available mixdown.  HandBrake will sanitize the default to an allowed mixdown that is compatible with the input channel layout.  So this should jiust return 7point1 here like the others. :wink:

Nomis101 commented:
OK, seems it worked. Thanks.

Nomis101 committed:
libhb: Allow 7.1 channels AAC encoding.

Allow 7.1 channels for AAC, but dissalow 6.1 for FDK (H)AAC, because it is not supported.

jstebbins commented:
LGTM

bradleysepos commented:
I’ll rebase and merge this in a bit.

bradleysepos commented:
Merged. Thanks @Nomis101!

Nomis101 commented:
Cool! Thanks for rebasing, merging and all the help!
